File: repose-aggregator/functional-tests/test-bundles/filter-one/src/main/java/org/openrepose/filters/one/FirstFilter/ClassLoaderServletRequestWrapper.java
package org.openrepose.filters.one.FirstFilter;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletRequestWrapper;
import org.apache.commons.lang3.StringUtils;


/**
 * Created by dimi5963 on 1/5/15.
 */
public class ClassLoaderServletRequestWrapper extends HttpServletRequestWrapper {
    public ClassLoaderServletRequestWrapper(HttpServletRequest request){
        super(request);
    }

    @Override
    public String getHeader(String s) {
        if(StringUtils.startsWith(s, "FOO")){
            s = new SimplicityDivine().createBar();
        }
        return super.getHeader(s);
    }


}

File: repose-support/filter-five/src/main/scala/org/openrepose/filters/core/filterfive/ClassLoaderServletRequestWrapper.scala
package org.openrepose.filters.core.filterfive

import javax.servlet.http.{HttpServletRequest, HttpServletRequestWrapper}

import org.apache.commons.lang3.StringUtils

import org.openrepose.others.SimplicityDivine

/**
 * Created by dimi5963 on 1/6/15.
 */
class ClassLoaderServletRequestWrapper(request: HttpServletRequest) extends HttpServletRequestWrapper(request) {
  override def getHeader(headerString: String): String = {
    if(StringUtils.startsWith(headerString, "FOO")){
      return new SimplicityDivine().createBar
    } else {
      return super.getHeader(headerString)
    }
  }
}
